File: src/04_quicksort/ts/04_recursive_max.ts
function maxRecursive(arr: number[], num: number): number {
  const { length } = arr;

  if (num == 1) {
    return arr[0];
  }
  // console.log(num);

  return Math.max(arr[num - 1], maxRecursive(arr, num - 1));
}
const A = [5, 5, 3, 6];
const n = A.length;
console.log(maxRecursive(A, n));

File: src/04_quicksort/python/04_recursive_max.py
def max_(arr: list[int]) -> (int | None):
    if len(arr) == 0:
        return None

    if len(arr) == 1:
        return arr[0]

    else:
        sub_max = max_(arr[1:])
        return arr[0] if arr[0] > sub_max else sub_max
